src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/CodeGen/ReadyToRunObjectWriter.cs

```diff
@@ -162,6 +164,16 @@ public void EmitPortableExecutable()
                         nativeDebugDirectoryEntryNode = nddeNode;
                     }
 
+                    if (node is ImportThunk importThunkNode)
+                    {
+                        // All the import thunks are in a single contiguous run
+                        if (firstImportThunk == null)
```

Comment:
Could you add a check to ensure that this is a contiguous run of imports nodes? This is the sort of thing that might break as we adjust sorting for improved memory layout, and I'd like to have the compiler checking for this.

Reply:
Done